main_test.go

```diff
@@ -787,27 +807,49 @@ func fakeCHHandler(w http.ResponseWriter, r *http.Request) {
 // Results are asserted according to the specified input parameters.
 func executeTwoConcurrentRequests(t *testing.T, query string, firstStatusCode, secondStatusCode int, firstBody, secondBody string) {
 	u := fmt.Sprintf("http://127.0.0.1:9090?query=%s&user=concurrent_user", url.QueryEscape(query))
-	req, err := http.NewRequest("GET", u, nil)
-	checkErr(t, err)
 
 	var wg sync.WaitGroup
 	wg.Add(2)
 	var resp1 string
 	var resp2 string
+	errs := make(chan error, 0)
+	defer close(errs)
+	errors := make([]error, 0)
+	go func() {
+		for err := range errs {
+			errors = append(errors, err)
+		}
+	}()
 	go func() {
-		resp := httpRequest(t, req, firstStatusCode)
+		defer wg.Done()
+		req, err := http.NewRequest("GET", u, nil)
+		checkErr(t, err)
+		resp, err := httpRequest(t, req, firstStatusCode)
+		if err != nil {
+			errs <- err
+			return
+		}
 		resp1 = bbToString(t, resp.Body)
-		wg.Done()
 	}()
 
 	go func() {
+		defer wg.Done()
 		time.Sleep(20 * time.Millisecond)
```

Comment:
why do we need to sleep for 20 msec. Our logic of conccurence query should work even if we don't sleep